File: algorithms/165.py
# https://leetcode.com/problems/compare-version-numbers


class Solution(object):
    def compareVersion(self, version1, version2):
        """
        :type version1: str
        :type version2: str
        :rtype: int
        """

        v1 = list(map(int, version1.split('.')))
        v2 = list(map(int, version2.split('.')))

        for i in range(max(len(v1), len(v2))):
            i1 = v1[i] if i < len(v1) else 0
            i2 = v2[i] if i < len(v2) else 0
            if i1 > i2:
                return 1
            elif i1 < i2:
                return -1

        return 0

File: algorithms/165.swift
// https://leetcode.com/problems/compare-version-numbers

class Solution {
    func compareVersion(version1: String, _ version2: String) -> Int {
        let v1 = version1.componentsSeparatedByString(".").map { Int($0) }
        let v2 = version2.componentsSeparatedByString(".").map { Int($0) }
        for i in 0..<max(v1.count, v2.count) {
            let i1 = (i < v1.count) ? v1[i] : 0
            let i2 = (i < v2.count) ? v2[i] : 0
            if i1 > i2 { return 1 }
            else if i1 < i2 { return -1 }
        }
        return 0
    }
}
